cdap-docs/developers-manual/source/building-blocks/datasets/index.rst

```diff
@@ -138,7 +141,23 @@ Tables. These Tables are similar to tables in a relational database with a few k
   and written independently of other columns, and columns are ordered
   in byte-lexicographic order. They are also known as *Ordered Columnar Tables*.
 
-|fileset|_ is a core Dataset type that was added as an experimental feature in CDAP 2.6.0.
+
+A FileSet represents a collections of files in the file system that share some common attributes
+such as the format and schema, while abstracting from the actual underlying file system interfaces.
+
+- Every file in a FileSet is in a location relative to the FileSet's base directory.
+
+- Knowing a file's relative path, any program can obtain a ``Location`` for that file through a method
+  of the FileSet dataset. It can then directly interact with the file's Location, for example write
+  data to the Location, or read data from it.
+
+- A FileSet can be used as the input or output to MapReduce. The MapReduce program need not specify
+  the input and output format to use, or configuration for these—the FileSet dataset provides this
+  information to the MapReduce runtime system.
+
+- An extension of FileSets, a PartitionedFileSet allows associating meta data (partitioning keys) with
```

Comment:
An extension of FileSets, PartitionedFileSets allow the associating of meta data (partitioning keys) with each file. The file can then be addressed through its meta data, removing the need for programs to be aware of actual file paths.